cmd/tpm-manager/tpm-manager.go

```diff
@@ -0,0 +1,506 @@
+	"k8s.io/kubernetes/pkg/client/leaderelection"
+	"k8s.io/kubernetes/pkg/client/record"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/flag"
+	nodeutil "k8s.io/kubernetes/pkg/util/node"
+	"k8s.io/kubernetes/pkg/util/tpm"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+// Flag a node as untrusted
+func invalidateNode(node *api.Node) error {
+	if node.ObjectMeta.Annotations == nil {
+		node.ObjectMeta.Annotations = make(map[string]string)
+	}
+        newTaints := []api.Taint{}
+        untrustedTaint := api.Taint{
+                Key:    "Untrusted",
+                Value:  "true",
+                Effect: api.TaintEffectNoSchedule,
+        }
+        taints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+        for _, taint := range taints {
+                if taint.Key == "Untrusted" {
+                        continue
+                }
+                newTaints = append(newTaints, taint)
+        }
+        newTaints = append(newTaints, untrustedTaint)
+        jsonContent, err := json.Marshal(newTaints)
+        node.Annotations[api.TaintsAnnotationKey] = string(jsonContent)
+        return err
+}
+
+func validateNode(node *api.Node) error {
+	if node.ObjectMeta.Annotations == nil {
+		node.ObjectMeta.Annotations = make(map[string]string)
+	}
+        newTaints := []api.Taint{}
+        taints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+        for _, taint := range taints {
+                if taint.Key == "Untrusted" {
+                        continue
+                }
+                newTaints = append(newTaints, taint)
+        }
+        jsonContent, err := json.Marshal(newTaints)
+        node.Annotations[api.TaintsAnnotationKey] = string(jsonContent)
+        return err
+}
+
+func isTrusted(node *api.Node) bool {
+	if node.ObjectMeta.Annotations == nil {
+		node.ObjectMeta.Annotations = make(map[string]string)
+	}
+	taints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+	if err != nil {
+		return false
+	}
+        for _, taint := range taints {
+                if taint.Key == "Untrusted" {
+			return false
+		}
+	}
+	return true
+}
+
+func getPolicy() (err error) {
+	manager.pcrconfigs = make([]map[string]tpm.PCRConfig, 0)
+
+	if manager.pcrconfig != "" {
+		pcrdata := make(map[string]tpm.PCRConfig)
+		pcrconfig, err := ioutil.ReadFile(manager.pcrconfig)
+		if err != nil {
+			glog.Errorf("Unable to read valid PCR configuration %s: %v", manager.pcrconfig, err)
+		}
+		err = json.Unmarshal(pcrconfig, &pcrdata)
+		if err != nil {
+			glog.Errorf("Unable to parse valid PCR configuration %s: %v", manager.pcrconfig, err)
+		}
+		for pcr, _ := range pcrdata {
+			pcrtmp := pcrdata[pcr]
+			pcrtmp.Source = manager.pcrconfig
+			pcrdata[pcr] = pcrtmp
```

Comment:
Actually if just replacing a string value, loop can probably be simplified to just:

```
for pcr := range pcrdata {
    pcrdata[pcr].Source = manager.pcrconfig
}
```